emma-language/src/main/scala/org/emmalanguage/compiler/Compiler.scala

```diff
@@ -52,80 +52,37 @@ trait Compiler extends AlphaEq
   /** Implicit types to be removed */
   lazy val implicitTypes: Set[u.Type] = API.implicitTypes
 
-  lazy val preProcess: Seq[TreeTransform] = Seq(
+  lazy val preProcess = TreeTransform("preProcess", Seq(
     Source.removeImplicits(implicitTypes),
     fixSymbolTypes,
     stubTypeTrees,
     unQualifyStatics,
     normalizeStatements,
     Source.normalize
-  )
+  ))
 
-  lazy val postProcess: Seq[TreeTransform] = Seq(
+  lazy val postProcess = TreeTransform("postProcess", Seq(
     TreeTransform("api.Owner.atEncl", api.Owner.atEncl),
     qualifyStatics,
     restoreTypeTrees
-  )
+  ))
 
   def pipeline(
     typeCheck: Boolean = false, withPre: Boolean = true, withPost: Boolean = true
   )(
     transformations: TreeTransform*
-  ): u.Tree => u.Tree = {
+  ): TreeTransform = {
 
     val bld = Seq.newBuilder[TreeTransform]
     //@formatter:off
     if (typeCheck) bld += TreeTransform("Compiler.typeCheck", (tree: u.Tree) => this.typeCheck(tree))
-    if (withPre)   bld ++= preProcess
+    if (withPre)   bld += preProcess
     bld ++= transformations
-    if (withPost)  bld ++= postProcess
+    if (withPost)  bld += postProcess
     //@formatter:on
     val steps = bld.result()
 
-    if (!printAllTrees) Function.chain(steps.map(_.xfrm))
-    else Function.chain(List(print) ++ steps.map(tt => executeAndPrintTransform(tt)))
-  }
-
-  implicit class TransformationOps(xfrm: TreeTransform) {
-    def iff(k: String)(implicit cfg: Config): Is =
-      new Is(k)
-
-    class Is(k: String)(implicit cfg: Config) {
-      def is(v: Boolean): TreeTransform =
-        if (cfg.getBoolean(k) == v) xfrm
-        else noop
-
-      def is(v: String): TreeTransform =
-        if (cfg.getString(k) == v) xfrm
-        else noop
-    }
-
-  }
-
-  val noop = TreeTransform("Predef.identity", Predef.identity[u.Tree] _)
-
-  // Turn this on to print the tree between every step in the pipeline (also before the first and after the last step).
-  val printAllTrees = false
```

Comment:
Are you removing the `printAllTrees` functionality intentionally? I often use it for debugging.

Reply:
The idea was to implement the functionality in another `XfrmEval` instance (similar to `TimingEval` or `TestEval`). Then you can just temporarily change the evaluator instance for debugging.

Reply:
OK, makes sense.